packages/chevrotain/src/parse/grammar/checks.ts

```diff
@@ -381,21 +381,20 @@ export function validateEmptyOrAlternative(
   topLevelRule.accept(orCollector)
   const ors = orCollector.alternations
 
-  const errors = reduce(
+  const errors = flatMap<Alternation, IParserEmptyAlternativeDefinitionError>(
```

Comment:
I feel as tough the technique of using `flatMap` to also `filter` is not the most readable.
(not that the previous code was very good either 😢 ).

If we are going to change this code block, could we do so in multiple steps which would be easier to understand?
- map to results of `nextPossibleTokensAfter` with additional info needed for buliding the error message later.
- filter out those for which: `isEmpty(possibleFirstInAlt)`
- map those to error objects

WDYT?

Reply:
As much as I'm a fan of function style myself, if readability were the priority, I would just do `const errors = []; for(...) { if (problem) errors.push(...) }`

Reply:
The imperative style you mentioned would be readable, same as the functional style I mentioned above.
The previous version (I originally wrote) and the proposed new versions are slightly less optimal.
I will try to improve those.

Reply:
but I have 31 other files to go over so no time to upgrade it. 😢 
At least it passes strict null check